File: backend/app/models/content.py
from sqlalchemy import Column, Integer, String, Text, DateTime, Float, ForeignKey, Enum
from sqlalchemy.sql import func
from datetime import datetime
from app.database import Base
import enum

class ContentType(str, enum.Enum):
    BLOG_POST = "blog_post"
    SOCIAL_MEDIA = "social_media"
    EMAIL = "email"
    NEWSLETTER = "newsletter"
    AD_COPY = "ad_copy"
    PRODUCT_DESCRIPTION = "product_description"

class ContentTone(str, enum.Enum):
    PROFESSIONAL = "professional"
    CASUAL = "casual"
    FUNNY = "funny"
    FORMAL = "formal"
    FRIENDLY = "friendly"

class Content(Base):
    __tablename__ = "contents"
    
    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    title = Column(String, nullable=False)
    content = Column(Text, nullable=False)
    content_type = Column(Enum(ContentType), default=ContentType.BLOG_POST)
    tone = Column(Enum(ContentTone), default=ContentTone.PROFESSIONAL)
    language = Column(String, default="en")
    
    # SEO
    keywords = Column(String, nullable=True)
    meta_description = Column(String, nullable=True)
    slug = Column(String, nullable=True, index=True)
    seo_score = Column(Float, default=0.0)
    
    # AI
    model_used = Column(String, default="gpt-4")
    tokens_used = Column(Integer, default=0)
    
    # Metadata
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
    
    def __repr__(self):
        return f"<Content {self.id}>"

class ContentTemplate(Base):
    __tablename__ = "content_templates"
    
    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False)
    description = Column(String, nullable=True)
    content_type = Column(Enum(ContentType), nullable=False)
    template = Column(Text, nullable=False)
    variables = Column(String, nullable=True)  # JSON string
    is_active = Column(Boolean, default=True)
    
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
    
    def __repr__(self):
        return f"<ContentTemplate {self.name}>"

File: frontend/lib/api/content.ts
import apiClient from '../api';

export interface ContentCreateRequest {
  title: string;
  content_type: 'blog_post' | 'social_media' | 'email' | 'newsletter' | 'ad_copy' | 'product_description';
  tone: 'professional' | 'casual' | 'funny' | 'formal' | 'friendly';
  language?: string;
  keywords?: string;
  prompt: string;
}

export interface Content {
  id: number;
  user_id: number;
  title: string;
  content: string;
  content_type: string;
  tone: string;
  language: string;
  keywords?: string;
  meta_description?: string;
  seo_score: number;
  model_used: string;
  tokens_used: number;
  created_at: string;
  updated_at: string;
}

export const contentAPI = {
  generate: async (data: ContentCreateRequest): Promise<Content> => {
    const response = await apiClient.post('/content/generate', data);
    return response.data;
  },

  getContents: async (skip: number = 0, limit: number = 10): Promise<Content[]> => {
    const response = await apiClient.get(`/content/?skip=${skip}&limit=${limit}`);
    return response.data;
  },

  getContent: async (id: number): Promise<Content> => {
    const response = await apiClient.get(`/content/${id}`);
    return response.data;
  },

  rewrite: async (contentId: number, tone?: string, style?: string) => {
    const response = await apiClient.post('/content/rewrite', {
      content_id: contentId,
      tone,
      style,
    });
    return response.data;
  },
};
